clang/tools/clang-linker-wrapper/ClangLinkerWrapper.cpp

```diff
@@ -599,9 +599,11 @@ Expected<StringRef> writeOffloadFile(const OffloadFile &File) {
 
   StringRef Prefix =
       sys::path::stem(Binary.getMemoryBufferRef().getBufferIdentifier());
-
-  auto TempFileOrErr = createOutputFile(
-      Prefix + "-" + Binary.getTriple() + "-" + Binary.getArch(), "o");
+  SmallString<128> Filename;
+  (Prefix + "-" + Binary.getTriple() + "-" + Binary.getArch())
+      .toNullTerminatedStringRef(Filename);
```

Comment:
```suggestion
      .toVector(Filename);
```
The return value of `toNullTerminatedStringRef` is not used. If there is a shortcut, `toNullTerminatedStringRef` doesn't even update the SmallVector, so this would potentially incorrect.

Reply:
I think this *technically* works, but you're right that `toVector` is better.

Reply:
```suggestion
SmallString<128> Filename{Prefix, "-", Binary.getTriple(), "-", Binary.getArch()};
```
should work as well